intelhex/__init__.py

```diff
@@ -693,14 +657,60 @@ def write_hex_file(self, f, write_start_addr=True, eolstyle='native'):
                     # adjust cur_addr/cur_ix
                     cur_ix += chain_len
                     if cur_ix < addr_len:
-                        cur_addr = addresses[cur_ix]
+                        if write_ela and (addresses[cur_ix] - cur_addr > 0x10):
```

Comment:
Do not hardcode ```0x10``` but use ```data_rec_len``` instead. Also check for that parameter in line 593 to get an ELA for shorter hex files.